Source/Core/Core/HW/MemmapFunctions.cpp

```diff
@@ -803,39 +719,31 @@ static __forceinline void UpdateTLBEntry(const XCheckTLBFlag _Flag, UPTE2 PTE2,
 	if (_Flag == FLAG_NO_EXCEPTION)
 		return;
 
-	PowerPC::tlb_entry *tlbe = PowerPC::ppcState.tlb[_Flag == FLAG_OPCODE][(vpa >> HW_PAGE_INDEX_SHIFT) & HW_PAGE_INDEX_MASK];
-	if ((tlbe[0].flags & TLB_FLAG_MOST_RECENT) == 0 || (tlbe[0].flags & TLB_FLAG_INVALID))
-	{
-		tlbe[0].flags = TLB_FLAG_MOST_RECENT;
-		tlbe[1].flags &= ~TLB_FLAG_MOST_RECENT;
-		tlbe[0].paddr = PTE2.RPN << HW_PAGE_INDEX_SHIFT;
-		tlbe[0].pte = PTE2.Hex;
-		tlbe[0].tag = vpa & ~0xfff;
-	}
-	else
-	{
-		tlbe[1].flags = TLB_FLAG_MOST_RECENT;
-		tlbe[0].flags &= ~TLB_FLAG_MOST_RECENT;
-		tlbe[1].paddr = PTE2.RPN << HW_PAGE_INDEX_SHIFT;
-		tlbe[1].pte = PTE2.Hex;
-		tlbe[1].tag = vpa & ~0xfff;
-	}
+	int tag = vpa >> HW_PAGE_INDEX_SHIFT;
+	PowerPC::tlb_entry *tlbe = &PowerPC::ppcState.tlb[_Flag == FLAG_OPCODE][tag & HW_PAGE_INDEX_MASK];
+	int index = tlbe->recent == 0 && tlbe->tag[0] != TLB_TAG_INVALID;
+	tlbe->recent = index;
```

Comment:
Okay, so you're saying in the case of a C update, this would be correct?

if (tlbe->tag[0] == tag)
	{
		// Check if C bit requires updating
		if (_Flag == FLAG_WRITE)
		{
			UPTE2 PTE2;
			PTE2.Hex = tlbe->pte[0];
			PTE2.C = 1;
			tlbe->pte[0] = PTE2.Hex;
		}

		if (_Flag != FLAG_NO_EXCEPTION)
			tlbe->recent = 0;

		*paddr = tlbe->paddr[0] | (vpa & 0xfff);

		return 1;
	}